gamemaker.py

```diff
@@ -30,4 +32,52 @@ def logout(self):
         return return_value
 
     def display_menu(self):
-        return "Options\n\nlog out\ndisplay status\n"
+        return "Options\n\nmake team\n\nlog out\ndisplay status\n"
+
+    def make_team(self, input):
+        input = input.split(" ")
+        if len(input) == 2:
+            name = input[0]
+            password = input[1]
+            self.database.add_team(team.Team(name, password, self.database))
+            ret_string = "Added " + name
+        else:
+            ret_string = "Invalid input!"
+        return ret_string
+
+    def edit_team(self, input):
+        found = False
+        input = input.split(" ")
+        if len(input) == 3:
+            list = self.database.get_teams()
+            for teams in list:
+                if teams.username == input[0]:
+                    found = True
+                    teams.username = input[1]
+                    teams.password = input[2]
+                    ret_string = "" + input[0] + " changed to " + input[1] + " with password " + input[2]
+        else:
+            ret_string = "Invalid input!"
+            found = True
+        if not found:
+            ret_string = "Could not find that team!"
+        return ret_string
+
+    def set_penalties(self, input):
+        exception = False
+        input = input.split(" ")
+        if len(input) == 2:
+            try:
```

Comment:
The handling of the exception should take place in the except block, not in the else block below

Reply:
I handled the exception with the boolean value. The "else" block below isn't handling the exception; its responding to the error that caused the exception.

Reply:
having exception = True is pointless. Should look like this:

try:
     time = int(input[1])
     guess = int(input[2])
     self.database.set_time_penalty(time)
     self.database.set_guess_penalty(guess)
     ret_string = "Time penalty is " + input[1] + " minutes and guess penalty is " + input[2] + " guesses"
except ValueError:
     ret_string = "Invalid input! Need integers greater than 0"

Reply:
Unless we want to accept negative values as penalty constraints, this won't work because negative values are integers. We need something that can reject both integers less than zero and non-integers.